src/nbl/ui/CClipboardManagerXCB.cpp

```diff
@@ -0,0 +1,147 @@
+
+#include "nbl/ui/CClipboardManagerXCB.h"
+#include "nbl/ui/XCBConnection.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include <xcb/xproto.h>
+
+namespace nbl::ui
+{
+    std::string CClipboardManagerXCB::getClipboardText() {
+        {
+            std::unique_lock<std::mutex> lk(m_clipboardMutex);
+            m_clipboardResponseCV.wait_until(lk, std::chrono::system_clock::now() + std::chrono::seconds(1));
+        }
+        std::lock_guard lk(m_clipboardMutex);
+        std::string response = std::move(m_clipboardResponse);
+        m_clipboardResponse = std::string();
+        return response;
+    }
+
+
+    bool CClipboardManagerXCB::setClipboardText(const std::string_view& data) {
+        std::lock_guard lk(m_clipboardMutex);
+        m_savedClipboard = data;
```

Comment:
arent you missing some event you should send or something?

Reply:
the clipboard is a mess ... I need to rethink this